sdk/python/generator/Generator.py

```diff
@@ -10,122 +10,89 @@
+			size_field_name = field_model.size
+			size_field_model = next(field2_model for field2_model in struct_model.fields if size_field_name == field2_model.name)
+			size_field_model.extensions.bound_field = field_model
 
-			struct_field.add_sizeof_field(field)
+		if field_model.is_size_reference:
+			struct_field_model = next(field2_model for field2_model in struct_model.fields if field_model.value == field2_model.name)
+			field_model.extensions.bound_field = struct_field_model
+			struct_field_model.extensions.size_fields.append(field_model)
 
 
-def process_struct(type_map, struct_type: StructObject, abstract_impl_map: AbstractImplMap):
-	if struct_type.ast_model.factory_type:
-		process_concrete_struct(type_map, struct_type, abstract_impl_map)
+def process_struct(type_map, struct_model, abstract_impl_map):
+	if struct_model.factory_type:
+		process_concrete_struct(type_map, struct_model, abstract_impl_map)
 
-	for field_ast_model in struct_type.ast_model.fields:
-		type_instance = type_map.get(field_ast_model.field_type, None)
+	for field_model in struct_model.fields:
+		type_model = type_map.get(field_model.field_type, None)
 
 		create_printer = create_pod_printer
-		if DisplayType.TYPED_ARRAY == field_ast_model.display_type:
-			element_type = type_map.get(field_ast_model.field_type.element_type, None)
-			if not element_type:
-				element_type = to_virtual_type_instance(field_ast_model.field_type.element_type)
-
-			type_instance = BaseObject(field_ast_model)
-			type_instance.is_contents_abstract = (
-				DisplayType.STRUCT == element_type.ast_model.display_type and element_type.ast_model.is_abstract
-			)
-		elif not type_instance:
-			type_instance = to_virtual_type_instance(field_ast_model)
+		is_contents_abstract = False
+		if DisplayType.TYPED_ARRAY == field_model.display_type:
+			element_type_model = type_map.get(field_model.field_type.element_type, None)
+			is_contents_abstract = DisplayType.STRUCT == element_type_model.display_type and element_type_model.is_abstract
+
+			type_model = field_model
+		elif not type_model:
+			type_model = field_model
 		else:
 			create_printer = BuiltinPrinter
 
-		field_printer = create_printer(type_instance, field_ast_model.name)
-		struct_type.add_field(Field(field_ast_model, type_instance, field_printer))
+		field_printer = create_printer(type_model, field_model.name)
+		field_model.extensions = AstExtensions(type_model, field_printer)
+		field_model.extensions.is_contents_abstract = is_contents_abstract
 
-	bind_size_fields(struct_type)
+	bind_size_fields(struct_model)
 
 
 def generate_files(ast_models, output_directory: Path):
 	# build map of types
-	type_map = {}
-	abstract_impl_map = AbstractImplMap()
-
-	types = []
-	for ast_model in ast_models:
-		instance = to_virtual_type_instance(ast_model)
-		types.append(instance)
-		type_map[ast_model.name] = instance
+	type_map = {ast_model.name: ast_model for ast_model in ast_models}
```

Comment:
not sure about ast_model naming in this method, but don't mind much, so up to you

Reply:
i don't mind it since it's called by the generate entry point that is using that naming